src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java

```diff
@@ -132,6 +133,31 @@ public String apply(RuleExecutionContext context) {
     }
   },
 
+  CONCAT_SUBFIELDS_BY_NAME() {
+    @Override
+    public String apply(RuleExecutionContext context) {
+      StringBuilder subfieldValue = new StringBuilder(context.getSubFieldValue());
+      DataField dataField = context.getDataField();
+      JsonArray subFields = (JsonArray) context.getRuleParameter().getValue("subfieldsToConcat");
+      int subFieldIndex = IntStream.range(0, dataField.getSubfields().size())
+        .filter(i -> dataField.getSubfields().get(i).getData().equals(context.getSubFieldValue()))
+        .findFirst()
+        .getAsInt();
+
+      for (int j = 0; j < subFields.size(); j++) {
```

Comment:
@bisecomsergiy  Could you please extract this logic to the method?

Reply:
Thanks Volodymyr, it is done!